extensions/wrapper/client/src/test/java/de/sovity/edc/client/ContractDefinitionPageApiServiceTest.java

```diff
@@ -0,0 +1,156 @@
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.client;
+
+import de.sovity.edc.client.gen.model.ContractDefinitionRequest;
+import de.sovity.edc.client.gen.model.CriterionDto;
+import org.eclipse.edc.connector.contract.spi.types.offer.ContractDefinition;
+import org.eclipse.edc.connector.spi.contractdefinition.ContractDefinitionService;
+import org.eclipse.edc.junit.annotations.ApiTest;
+import org.eclipse.edc.junit.extensions.EdcExtension;
+import org.eclipse.edc.spi.query.Criterion;
+import org.eclipse.edc.spi.query.QuerySpec;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ApiTest
+@ExtendWith(EdcExtension.class)
+class ContractDefinitionPageApiServiceTest {
+
+    @BeforeEach
+    void setUp(EdcExtension extension) {
+        TestUtils.setupExtension(extension);
+    }
+
+    @Test
+    void contractDefinitionPage(ContractDefinitionService contractDefinitionService) {
+        //arrange
+        var client = TestUtils.edcClient();
+        var criteria = Arrays.asList(
+                new Criterion(
+                        "exampleLeft1",
+                        "operator1",
+                        "exampleRight1")
+        );
+        createContractDefinition(contractDefinitionService, "contractPolicy-id-1", "accessPolicy-id-1", criteria);
+
+        // act
+        var result = client.uiApi().contractDefinitionPage();
+
+        //assert
+        var contractDefinitions = result.getContractDefinitions();
+        assertThat(contractDefinitions).hasSize(1);
+        var contractDefinition = contractDefinitions.get(0);
+        assertThat(contractDefinition.getContractPolicyId()).isEqualTo("contractPolicy-id-1");
+        assertThat(contractDefinition.getAccessPolicyId()).isEqualTo("accessPolicy-id-1");
+        assertThat(contractDefinition.getCriteria().get(0).getOperandLeft()).isEqualTo(criteria.get(0).getOperandLeft());
+        assertThat(contractDefinition.getCriteria().get(0).getOperator()).isEqualTo(criteria.get(0).getOperator());
+        assertThat(contractDefinition.getCriteria().get(0).getOperandRight()).isEqualTo(criteria.get(0).getOperandRight());
+    }
+
+    @Test
+    void contractDefinitionPageSorting(ContractDefinitionService contractDefinitionService) {
+        //arrange
+        var client = TestUtils.edcClient();
+        createContractDefinition(contractDefinitionService, "contractPolicy-id-1", "accessPolicy-id-1", Arrays.asList());
+        TestUtils.wait(1);
+        createContractDefinition(contractDefinitionService, "contractPolicy-id-2", "accessPolicy-id-2", Arrays.asList());
+        TestUtils.wait(1);
```

Comment:
Question: Why do we need to wait? (seconds are a long time)

Reply:
CDs are occasionally created out of order, disrupting the assert operation. By introducing a wait, we ensure sequential creation of each contract definition.